person.rb

```diff
@@ -1,18 +1,31 @@
 require 'securerandom'
-
-class Person
+require_relative './nameable'
+require_relative './capitalize_decorator'
+require_relative './trimmer_decorator'
+require_relative './rental'
+class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = SecureRandom.uuid
+    @rentals = []
     @name = name
     @age = age
-    @parent_permission = parent_permission
+    @parent_permission = !parent_permission.nil?
+  end
+
+  def add_rental(book)
+    Rental.new(book, self)
```

Comment:
- Please, let's also add a `date` parameter to both parentheses so that we get to set the date when a book is rented out.

Reply:
* This requested change has not been fixed, as mentioned above kindly add a date parameter to your add_rental method